include/lsst/afw/image/VisitInfo.h

```diff
@@ -0,0 +1,217 @@
+
+    bool operator==(VisitInfo const & other) const;
+    bool operator!=(VisitInfo const & other) const { return !(*this == other); };
+
+    /// get exposure ID
+    table::RecordId getExposureId() const { return _exposureId; }
+
+    /// get exposure duration (sec)
+    double getExposureTime() const { return _exposureTime; }
+
+    /// get uniform date and time at middle of exposure
+    daf::base::DateTime getDate() const { return _date; }
+
+    /// get UT1 (universal time) MJD date at middle of exposure
+    double getUt1() const { return _ut1; }
+
+    /// get earth rotation angle at middle of exposure
+    geom::Angle getEra() const { return _era; }
+
+    /// get ICRS RA/Dec position at the boresight
+    /// (and at the middle of the exposure, if it varies with time)
+    coord::IcrsCoord getBoresightRaDec() const { return _boresightRaDec; }
+
+    /// get refracted apparent topocentric Az/Alt position at the boresight
+    /// (and at the middle of the exposure, if it varies with time)
+    coord::Coord getBoresightAzAlt() const { return _boresightAzAlt; }
+
+    /// get airmass at the boresight, relative to zenith at sea level
+    /// (and at the middle of the exposure, if it varies with time)
+    double getBoresightAirmass() const { return _boresightAirmass; }
+
+    /**
+    * Get rotation angle at boresight at middle of exposure
+    *
+    * Rotation angle is angle of coordinate system specified by boresightRotType with respect to detector.
+    * For example if boresightRotType is SKY then at a rotation angle of 0 North is along detector Y axis
+    * (never flipped) and East is along the detector X axis (flipped if necessary)
+    */
+    geom::Angle getBoresightRotAngle() const { return _boresightRotAngle; }
+
+    /// get rotation type of boresightRotAngle
+    RotType getRotType() const { return _rotType; }
+
+    /// get observatory longitude, latitude and elevation
+    coord::Observatory getObservatory() const { return _observatory; }
+
+    /// get basic weather information
+    coord::Weather getWeather() const { return _weather; }
+
+    bool isPersistable() const { return true; }
+
+protected:
+
+    virtual std::string getPersistenceName() const;
+
+    virtual void write(OutputArchiveHandle & handle) const;
+
+private:
+    table::RecordId _exposureId;
+    double _exposureTime;
+    daf::base::DateTime _date;
+    double _ut1;
+    geom::Angle _era;
+    coord::IcrsCoord _boresightRaDec;
+    coord::Coord _boresightAzAlt;
+    double _boresightAirmass;
+    geom::Angle _boresightRotAngle;
+    RotType _rotType;
+    coord::Observatory _observatory;
+    coord::Weather _weather;
+};
+
+namespace detail {
+
+/**
+ * Set FITS metadata from a VisitInfo
+ *
+ * @param[in,out] metadata  FITS keyword metadata to set
+ * @param[in] visitInfo  instance of VisitInfo from which to set metadata
+ */
+void setVisitInfoMetadata(daf::base::PropertyList &metadata, VisitInfo const &visitInfo);
```

Comment:
I copied how Calib did it. I don't have strong feelings, but prefer to keep the API simple and use external free functions when it makes sense to do so, and FITS arcana may be such a situation.